packages/chrome/src/ChromeProvider/ChromeProvider.tsx

```diff
@@ -7,28 +9,70 @@ import { IDENTITY_URL, LAST_VISITED_URL, get, post } from '../utils/fetch';
 
 const getUserIdentity = () => get<UserIdentity>(IDENTITY_URL);
 
-const useLastPageVisitedUploader = (providerState: ReturnType<typeof chromeState>, bundle = '') => {
+const useLastPageVisitedUploader = (providerState: ReturnType<typeof chromeState>) => {
+  const scalprum = useScalprum<{ initialized: boolean; api: { chrome: ChromeAPI } }>();
   const { pathname } = useLocation();
+  const postData = async (pathname: string, title: string, bundle: string) => {
+    try {
+      const data = await post<LastVisitedPage[], { pathname: string; title: string; bundle: string }>(LAST_VISITED_URL, {
+        pathname,
+        title,
+        bundle,
+      });
+      providerState.setLastVisited(data);
+    } catch (error) {
+      console.error('Unable to update last visited pages!', error);
+    }
+  };
   useEffect(() => {
-    post<LastVisitedPage[], { pathname: string; title: string; bundle: string }>(LAST_VISITED_URL, {
-      pathname,
-      title: document.title,
-      bundle,
-    })
-      .then((data) => providerState.setLastVisited(data))
-      .catch((error) => console.error('Unable to update last visited pages!', error));
+    let titleObserver: MutationObserver | undefined;
+    let prevTitle: string | null;
+    const titleTarget = document.querySelector('title');
+    if (titleTarget) {
+      prevTitle = titleTarget.textContent;
+      // initial api call on mount
+      postData(pathname, prevTitle ?? '', scalprum.api.chrome.getBundleData().bundleTitle);
+      /**
+       * Use Mutation observer to trigger the updates.
+       * Using the observer will ensure the last visited pages gets updated on document title change rather than just location change.
+       * The chrome service uses pathname as identifier and updates title according.
+       * Multiple calls with the same pathname and different title will ensure that the latest correct title is assigned to a pathname.       *
+       *  */
+      titleObserver = new MutationObserver((mutations) => {
+        // grab text from the title element
+        const currentTitle = mutations[0]?.target.textContent;
+        console.warn({ prevTitle, currentTitle, pathname });
```

Comment:
Do we have any mechanism to gate this to "debug" only scenarios?

Reply:
No, and it was added just for testing because the MutationObserver does not exist in node env. It should be removed.